add NOTE about failed builds related to CVE-2022-24765 (#55)
bisdn/bisdn-linux: README.md

* `full`: include the full BISDN Linux system, including baseboxd and OF-DPA.

> > **_NOTE:_** When trying to build a release prior to 4.6.1, please read the
> > [Known Issues](#known-issues) section regarding CVE-2022-24765 before
> > starting.

Reviewer: No idea if this works, but I was thinking about something like this:

```suggestion
> **_NOTE:_** When trying to build a release prior to 4.6.1, execute the following git command before building:
> ```
> git config --global safe.directory '*'
> ``` 
> to avoid build errors with packages using python-bpr.
>
> See [Known Issues](#known-issues) section regarding CVE-2022-24765 for details.
```

The most important information should come first, i.e. what you need to do. The details of why can come later.

* `full`: include the full BISDN Linux system, including baseboxd and OF-DPA.

> > **_NOTE:_** The git CVE-2022-24765 directly affects all builds of BISDN Linux

Reviewer: To me it interrupts the flow quite a bit, due to being a big note.

I would suggest splitting it into two:

Add a short note about setting safe.directory to '*' when building releases before 4.6.1 with the appropriate command in `## Bootstrap build system`.

And then add a section at the bottom named FAQ or known build issues with the error explained in detail (and maybe link the bootstrap part to there, "see X for details").

Author: My intention was to interrupt the flow as much as possible for now to avoid people first running a built for about 2 hours and only after it crashes reading the "Issues" section.

Reviewer: Setting this value even on older git version is fine (and will just get ignored there), so I think its enough to say "if you build for older versions, run this command first, and see here for an explanation/details".

Author: done

> prior to version 4.6.1, since those were release before the required patches
> were released in yocto itself:
>
>   - https://git.yoctoproject.org/poky/commit/?h=dunfell&id=90cf135b046e4195512daeb8e8f516603d0f5d6c
>   - https://git.yoctoproject.org/poky/commit/?h=dunfell&id=3a9cef8dbe3b5d81521536232c4ac67a9e81ac9a
>   - https://git.yoctoproject.org/poky/commit/?h=dunfell&id=a75678145b35b3722f0cbba51aa699d3fd8f0bef
>   - https://git.yoctoproject.org/poky/commit/?h=dunfell&id=a48231b5bfb5b1ca5de6dad6b3277a051eb6b55c
>
> Using a patched git version with an unpatched yocto one will cause builds to
> fail with an error message like:
>
> ```

> > Summary: 1 task failed:

Reviewer: Maybe add at least part of the actual error message here? So you can check that this is the right error you are looking at.

Author: I thought about it, but imho this part is the central message and contains most of the import key words.

Reviewer: Yeah, but it doesn't say why it failed, or what part of the install failed. The install can fail for various reasons, so there really should be more context.

Author: done

## Install image

Please refer to our [BISDN Linux docs](https://docs.bisdn.de/getting_started/install_bisdn_linux.html) on how to install the resulting image.

> ## Known Issues

Reviewer: Since we can now drop the requirment of having python2 and python2-yaml installed, we will want to add more to this sections in the future.

```suggestion
## Known Issues when building older versions of BISDN Linux
```

  | ERROR: 'python3 setup.py install --root=/tmp/accton-as4610/work/cortexa9-vfp-poky-linux-gnueabi/python3-oslo.i18n/3.17.0+gitAUTOINC+f2729cd36f-r0/image     --prefix=/usr     --install-lib=/usr/lib/python3.8/site-packages     --install-data=/usr/share' execution failed.
  | WARNING: exit code 1 from a shell command.
  | ERROR: Execution of '/tmp/accton-as4610/work/cortexa9-vfp-poky-linux-gnueabi/python3-oslo.i18n/3.17.0+gitAUTOINC+f2729cd36f-r0/temp/run.do_install.2518342' failed with exit code 1
  ERROR: Task (/home/ubuntu/workspace/poky-bisdn-linux/poky/meta-cloud-services/meta-openstack/recipes-devtools/python/python3-oslo.i18n_git.bb:do_install) failed with exit code '1'
  NOTE: Tasks Summary: Attempted 4488 tasks of which 8 didn't need to be rerun and 1 failed.
  
  Summary: 1 task failed:
    /home/ubuntu/workspace/poky-bisdn-linux/poky/meta-cloud-services/meta-openstack/recipes-devtools/python/python3-oslo.i18n_git.bb:do_install
  Summary: There were 25 WARNING messages shown.
  Summary: There were 2 ERROR messages shown, returning a non-zero exit code.
  ```

> 

Reviewer: ```suggestion

This is caused by the fix for git **CVE-2022-24765**(<insert link here>), which breaks the way Yocto handles building packages. A fix for Yocto was introduced after the release of BISDN Linux 4.6.0.
```

We should definitely link to the description of the CVE, so the user actually knows what this is.

If you do want to add links to the commits fixing it, I'd suggest using a footnote at the end, or [1] [2] [3] [4] as link titles to keep it small.

  To avoid this you can either try building from version 4.6.1 (or newer), or
  workaround the git safeguard and manually mark all directories as safe to
  built from by running:

  ```
  git config --global safe.directory '*'

>   ```

Reviewer: We should warn about the effect of this line:

```
  **_WARNING_**: This effectively reintroduces the vulnerability of CVE-2022-24765.
```

(can't use a suggestion here, because the ``` of the last line break suggesting stuff here)
